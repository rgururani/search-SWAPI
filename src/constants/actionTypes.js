export const USER_AUTHENTICATED = 'USER_AUTHENTICATED';
export const Planet_FETCH_SUCCESS = 'Planet_FETCH_SUCCESS';
export const GET_USER = 'GET_USER';
