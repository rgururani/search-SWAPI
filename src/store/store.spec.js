import * as ActionTypes from '../constants/actionTypes';
import { createStore } from 'redux';

import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

describe('Store', () => {

  it('should display results when necessary data is provided', () => {
    const store = createStore(rootReducer, initialState);

    const actions = [
    ];
    actions.forEach(action => store.dispatch(action));

    const actual = store.getState();
    const expected = {
      user : '',
      planets: [],
      routing: {
        locationBeforeTransitions:null
      }

    };

    expect(actual).toEqual(expected);
  });

  it('should not display results when necessary data is not provided', () => {
    const store = createStore(rootReducer, initialState);

    const actions = [
    ];

    actions.forEach(action => store.dispatch(action));

    const actual = store.getState();

    const expected = {
      user : '',
      planets: [],
      routing: {
        locationBeforeTransitions:null
      }
    };


    expect(actual).toEqual(expected);
  });


  it('should handle a flurry of actions', () => {
    const store = createStore(rootReducer, initialState);

    const actions = [

    ];
    actions.forEach(action => store.dispatch(action));

    const moreActions = [

    ];

    moreActions.forEach(action => store.dispatch(action));

    const actual = store.getState();
    const expected = {
      user : '',
      planets: [],
      routing: {
        locationBeforeTransitions:null
      }
    };


    expect(actual).toEqual(expected);
  });
});
