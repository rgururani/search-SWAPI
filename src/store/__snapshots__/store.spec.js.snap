// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store should handle a flurry of actions 1`] = `
Object {
  user : {

  },
  planets: [{

}]
}
`;
